odometry:
    data_dir: "/home/neo/data_maicity/mai_city/bin/sequences/01"
#    data_dir: "/home/neo/data_kitti/00"
    dataset_num: 2     # 1-kitti 2-maicity
    lid_topic:  "/cloud_registered"
    pose_topic: "/aft_mapped_to_init"

mesh:
    hole_every_n_frame: 1               # 每隔 n 帧一次hole_process
    angle_resolution: 0.4
    fov_horizon: 360
    fov_vertical_up: 2.0
    fov_vertical_down: 30
    r_kNN: 5                            # 最近邻计算特征值的框的边长
    depth_th_normal: 0.5                # 框内 点-中心点深度差 阈值
    eigenvalue_threshold: 0.0001        # 确定初始平面点的阈值
    angle_threshold: 5                  # 区域生长 角度相似性阈值 单位度
    depth_threshold: 0.01               # 区域生长 平面点法向距离阈值 单位m
    angle_th_spd: 5                     # 相似平面检测，法向角度阈值
    dis_th_spd: 0.03                    # 相似平面检测，沿法向距离阈值
    ptsNum_threshold: 40                # 平面点数量阈值
    dis_resolution: 0.15                 # 平面划分网格的边长
    grid_ptsNum_threshold: 0            # 确定网格为内部点的点数阈值
    ptsNum_decimate_threshold: 1000     # 点投影矩阵是否重新计算
    voxel_resolution: 0.3               # 非平面部分 体素边长
    points_minimum_scale: 0.05           # 非平面部分 任意两个点之间最小距离
    r_nonplane: 0.15                     # 非平面部分 最近邻查找距离阈值
    N_nonplane: 5                       # 非平面部分 拟合三角形的点数阈值
    max_pts_num: 15000                  # 原始点云数量整除该数进行降采样 TODO
    if_save_raw_points: false
    if_quadtree_decimate: true          # 是否进行四叉树简化
    if_gui: true                        # 是否显示gui
    if_debug: false
    ptcl_topic: "/cloud_registered"
    odo_topic: "/aft_mapped_to_init"
    ptcl_save_path: "/home/neo/output_onlinemesh"
    dataset_name: "maicity"
